bofire/data_models/features/categorical.py

```diff
@@ -14,14 +13,18 @@
     TCategoryVals,
     TTransform,
 )
+from bofire.data_models.objectives.categorical import (
+    CategoricalObjective,
+    ConstrainedCategoricalObjective,
+)
 
 
 class CategoricalInput(Input):
     """Base class for all categorical input features.
 
     Attributes:
-        categories (List[str]): Names of the categories.
-        allowed (List[bool]): List of bools indicating if a category is allowed within the optimization.
+        categories (Tuple[str]): Names of the categories.
```

Comment:
Let's do this in a later PR with changing the types here from list to tuple, as mentioned above. It could then even make sense to have it not as tuple but as set.